app/src/main/java/com/nutomic/syncthingandroid/activities/WebGuiActivity.java

```diff
@@ -141,7 +141,21 @@ public void onServiceConnected(ComponentName componentName, IBinder iBinder) {
 
     @Override
     public void onWebGuiAvailable() {
-        mWebView.loadUrl(getService().getWebGuiUrl().toString());
+        final String serviceUrl = getService().getWebGuiUrl().toString();
+        if (!isCurrentUrl(serviceUrl)) {
+            mWebView.loadUrl(serviceUrl);
+        }
+    }
+
+    private boolean isCurrentUrl(final String url) {
+        String currentUrl = mWebView.getUrl();
+        if (currentUrl == null || currentUrl.length() == 0) {
+            return false;
+        }
+        if (currentUrl.endsWith("/")) {
+            currentUrl = currentUrl.substring(0, currentUrl.length() - 1);
+        }
+        return currentUrl.equals(url);
     }
```

Comment:
Is there any reason for this check? I think it should be enough to check `mWebView.getUrl() == null` in onWebGuiAvailable().

Reply:
Also thanks for finding this, I wasn't aware that it was causing the reloads.

Reply:
well, I was not sure that the URL would always be null (or empty) as I am not familiar with syncthing's web API, so a check against the actual URL made sense.

Reply:
Ah I see. We actually only need to set the URL the a single time (to open the Syncthing web gui). So it would be enough to check if the current URL is null, and load the web gui in that case.

Reply:
In that case the changes in this pull request are a bit of an overkill. Should I create a separate, simplified pull request with a minimal check?